types/react-table/index.d.ts

```diff
@@ -305,6 +305,7 @@ export interface UseTableRowProps<D extends object> {
     original: D;
     id: string;
     subRows: Array<Row<D>>;
+    depth: number;
```

Comment:
I think that this is wrong.

`depth` is only valid if you are using the `useGroupBy` plugin, in which case you should extend the `Row` interface with `UseGroupByRowProps` which will add the `depth` property.

Reply:
@ggascoigne That's weird, the only plugin that I'm using is `useExpanded` and `depth` was given in `row` property. I'm going to double-check shortly

Reply:
![image](https://user-images.githubusercontent.com/8914311/104805986-6d872e80-5817-11eb-9cf1-bc0d3f2404b8.png)
yeah.. they do have `depth` property when `useExpanded` plugins is being used, so should I move it to the `UseExpandedRowProps` interface then?

Reply:
and... sorry for messing with PR but I really have no idea where to ask, it seems the [official example](https://react-table.tanstack.com/docs/examples/data-driven-classes-and-styles) doesn't work with current type declaration, so I just want to know how to make it work with my extended custom local type declaration

Reply:
Ah that makes sense, yes I think that adding it (not moving it) to `UseExpandedRowProps` would make sense.  As for an example - you could take a look at https://github.com/ggascoigne/react-table-example, it might help and uses the latest version of both react-table and the types.

Reply:
@ggascoigne Oh yes I've already checked that out and that helped me a lot really. But I think still there's some part that your example aren't covering yet, e.g)

Considering this pattern:
```jsx
  <tr {...row.getRowProps(getRowProps(row))}>
```
From my understanding, there is no way around passing object to the `getRowProps ` with current declaration.
but js implementation is somehow merging that object and convey it to the `<TableRow />` component
```ts
getRowProps: (propGetter?: RowPropGetter<D>) => TableRowProps;
```

and similarly, this also causes type error when you compile:

```jsx
  <th
                // Return an array of prop objects and react-table will merge them appropriately
                {...column.getHeaderProps([
                  {
                    className: column.className,
                    style: column.style,
                  },
                  getColumnProps(column),
                  getHeaderProps(column),
                ])}
              >
                {column.render('Header')}
              </th>
```
There's a good change that I'm missing something as I'm not that familiar with advanced type system, so feel free to correct me if I'm wrong